### ISSUE TITLE
vkDestroyQueryPool false positive

### ISSUE THREAD

**Issue by pc-john** [KhronosGroup/Vulkan-ValidationLayers#3332]
I am getting validation error that I believe is false positive in certain conditions:

VUID-vkDestroyQueryPool-queryPool-00793(ERROR / SPEC): msgNum: 1593032491 - Validation Error: [ VUID-vkDestroyQueryPool-queryPool-00793 ] Object 0: handle = 0x2aa01fbed28, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x5ef3bf2b | Cannot call vkDestroyQueryPool on VkQueryPool 0xfab64d0000000002[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to queryPool must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.189.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyQueryPool-queryPool-00793)

My scenario is this:
1. record command buffer using the queryPool with single vkCmdWriteTimestamp() call
2. submit the command buffer
3. call vkGetQueryPoolResults() to wait for the timestamp to be written
4. destroy queryPool  <- false positive reported here
5. wait for the command buffer execution to complete

I get validation error on step 4, although "The Vulkan spec states: All submitted commands that refer to queryPool must have completed execution." And they really completed the execution. The command buffer is still in the process of execution, but everything related to the queryPool has finished, all commands related to queryPool. Attaching a minimal example to reproduce the problem.
[VulkanQueryPoolDestroyValidationError.zip](https://github.com/KhronosGroup/Vulkan-ValidationLayers/files/7225902/VulkanQueryPoolDestroyValidationError.zip)

**Comment by ziga-lunarg** [comment]
> call vkGetQueryPoolResults() to wait for the timestamp to be written

vkGetQueryPoolResults() does not wait, it can return VK_NOT_READY. I don't think this is a false positive.

**Comment by pc-john** [comment]
Yes. But I am waiting until it returns VK_SUCCESS. And VK_SUCCESS means that all the execution related to the command pool completed. So, destroying it should not be a problem and no validation error shall be printed, in my opinion.

**Comment by ncesario-lunarg** [comment]
This might be a question for Vulkan-Docs. The VUID states:
> All submitted **commands** that refer to queryPool must have completed execution

The keyword (for me) is _commands_. While a return value from `vkGetQueryPoolResults` of `VK_SUCCESS` may guarantee the query is available, I'm not seeing in the spec (it's quite possible I'm just missing it) where it states that this also guarantees that the _command_ has finished. That seems like a perfectly good assumption, but an assumption nonetheless. From what I can see in the spec, some sort of synchronization method (e.g., fence, semaphore) must be used to ensure all command(s) (sounds like `vkCmdWriteTimestamp` in this case) have finished before calling `vkDestroyQueryPool`.

**Comment by ziga-lunarg** [comment]
It was asked here: https://github.com/KhronosGroup/Vulkan-Docs/issues/1647
